### PR TITLE
LibCore: add format option for DateTime::to_string()

### PR DESCRIPTION
**Pull request by howar6hill**
Add strftime style format options.
It's nice to be able to print time in different formats.

### CODE REVIEW

File: Libraries/LibCore/DateTime.cpp [hunk 1]
```diff
@@ -51,9 +50,15 @@ DateTime DateTime::from_timestamp(time_t timestamp)
     return dt;
 }
 
-String DateTime::to_string() const
+String DateTime::to_string(const String& format) const
 {
-    return String::format("%04u-%02u-%02u %02u:%02u:%02u", m_year, m_month, m_day, m_hour, m_minute, m_second);
+    struct tm tm;
+    localtime_r(&m_timestamp, &tm);
+    char buff[256];
```
Comment (awesomekling) [inline Libraries/LibCore/DateTime.cpp]: Why 256? This seems very arbitrary. Can we organize the code to allow any length?

Comment (awesomekling) [reply Libraries/LibCore/DateTime.cpp]: Also please capitalize commit messages :)

Comment (howar6hill) [reply Libraries/LibCore/DateTime.cpp]: As  we must pass a buffer size to the C function `strftime()`, if we want to allow arbitrary size, we 
can't call it directly,  and have to copy large parts of code from `strftime()`.

**Comment by howar6hill** [comment]
@awesomekling Modified to let `DateTime::to_string()` return arbitrary length string, but have to copy large parts of code from `strftime()`.

**Comment by awesomekling** [comment]
Oh sorry, I should have been more clear. I meant, is there some way we can share the code between these two things? Perhaps LibC could have a common helper function that we can use from both `strftime()` and `DateTime::to_string()`?

**Comment by howar6hill** [comment]
It could be done. Where do you think is the appropriate place to declare this helper function?  In `time.h`?

**Comment by howar6hill** [comment]
@awesomekling I want to create a `time_utils.h` in `LibC`, and put the implementation in `time_utils.cpp`. Is that OK?

**Comment by awesomekling** [comment]
> @awesomekling I want to create a `time_utils.h` in `LibC`, and put the implementation in `time_utils.cpp`. Is that OK?

Argh, actually, now that I think about this. We'll be making `LibCore` less portable by making it rely on a non-standard LibC mechanism.

I think actually your version with the duplication is best for now. Let's commit it that way. I'll merge this. Thank you for working on it :)

**Pull request merged**